src/Microsoft.OData.Core/Evaluation/ODataConventionalEntityMetadataBuilder.cs

```diff
@@ -230,6 +231,15 @@ internal override ODataStreamReferenceValue GetMediaResource()
             {
                 this.computedMediaResource = new ODataStreamReferenceValue();
                 this.computedMediaResource.SetMetadataBuilder(this, /*propertyName*/ null);
+
+                // from OData spec: Media entity types MAY specify a list of acceptable media types using an annotation with term Core.AcceptableMediaTypes
+                IEdmEntityType entityType = this.ResourceMetadataContext.ActualResourceType as IEdmEntityType;
+                var mediaTypes = this.MetadataContext.Model.GetVocabularyStringCollection(entityType, CoreVocabularyModel.AcceptableMediaTypesTerm);
+                if (mediaTypes != null)
+                {
+                    // TODO: Make sure we use ',' to concatant the multiple media types or pick the first one?

```

Comment:
List doesn't make sense for the returned ContentType.
This should use the first item.
So the convention behavior could only work when the acceptable types is a collection of one, which will be a fairly common case.

Reply:
@garethj-msft & @mikepizzo 

OData spec says:  http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateStreamValues:

*Stream properties MAY specify a list of acceptable media types using an annotation with term Core.AcceptableMediaTypes, see [OData-VocCore].*

http://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_MediaEntityType
: 
*Media entity types MAY specify a list of acceptable media types using an annotation with term Core.AcceptableMediaTypes, see [OData‑VocCore].*

The AcceptableMediaTypes is defined as **COLLECTION** of media type. 
```xml
<Term Name="AcceptableMediaTypes" Type="Collection(Edm.String)" Nullable="false" AppliesTo="EntityType Property Term" />
```

But, there's no **Executable** instruction about how to format it into "ContentType" of stream (entity/property).

Reply:
AcceptableMediaTypes should return a collection of media types, which is the set of media types that the media entity stream/stream property may contain.  If this collection contains a single media type (which is likely to be a common case), then we should set the Content-Type to be that single media type.  As Gareth says, we should not attempt to set Content-Type to multiple mediatypes; that's invalid as per the definition of the Content-Type header.

---
In reply to: [570412053](https://github.com/OData/odata.net/pull/1995#discussion_r570412053) [](ancestors = 570412053)